### DISCUSSION TITLE
Beginner Question: Exporting file from Lima to Home

### DISCUSSION THREAD

**Discussion by simoneae** [lima-vm/lima#996] [General]
Hello. I can copy (cp) a file from my home computer into lima, but I have not be able to export with cp. I tried copying the text file from lima to a directory on my home computer and to an empty text file on my home computer with full read/write/execute access. I have been using full paths but receive the following ERROR (cp: cannot create regular file 'path/to/file': Read-only file system) 

How can I send a copy of my file out of lima environment to use on my home computer? 
Thank you, 
Simone

**Comment by jandubois** [comment]
I guess it means you are trying to copy the file to a location you don't have write access to?

Working example:

```console
$ limactl cp default:/etc/passwd /tmp/passwd
passwd                                          100% 1682     2.0MB/s   00:00

$ head -3 /tmp/passwd
root:x:0:0:root:/root:/bin/bash
daemon:x:1:1:daemon:/usr/sbin:/usr/sbin/nologin
bin:x:2:2:bin:/bin:/usr/sbin/nologin
```

You can also exfiltrate files by copying them inside the VM to `/tmp/lima`, which has been mounted from your host:

```console
$ limactl shell default cp /etc/passwd /tmp/lima/

$ head -3 /tmp/lima/passwd
root:x:0:0:root:/root:/bin/bash
daemon:x:1:1:daemon:/usr/sbin:/usr/sbin/nologin
bin:x:2:2:bin:/bin:/usr/sbin/nologin
```

**Reply by simoneae** [reply]
Thank you, the exfiltrating file solution worked. Have a lovely day.
